In [1]:
import numpy as np
import pandas as pd

In [2]:
amazon_df = pd.read_csv(r'rough_data\amazon_dataset\amzn_all_items_reviews_dups_removed.csv', header=None)

In [3]:
amazon_df

0           1    2           3
0         A3AF8FFZAZYNE5  0000000078  5.0  1092182400
1          AH2L9G3DQHHAJ  0000000116  4.0  1019865600
2         A2IIIDRK3PRRZY  0000000116  1.0  1395619200
3         A1TADCM7YWPQ8M  0000000868  4.0  1031702400
4          AWGH7V0BDOJKB  0000013714  4.0  1383177600
...                  ...         ...  ...         ...
82677126  A349MZLDSQAIQC  BT00IU6O8K  3.0  1375401600
82677127  A1NFN9O4A2Q92V  BT00IU6O8K  5.0  1375142400
82677128  A1DA2L6U0AFMLL  dp-g310/do  5.0  1195948800
82677129   AIDHT6KXOL64T  SMLRBIMX03  5.0  1355788800
82677130  A161I1MHX2G2BR  SMLRBIMX03  4.0  1364860800

[82677131 rows x 4 columns]

This is a huge dataset, 82.6M rows! Thankfully the authors managed to reduce the columns in this version, the original with all review text and metadata was 20gb!

In [5]:
amazon_df.columns = ['user', 'item', 'rating', 'timestamp']
amazon_df

user        item  rating   timestamp
0         A3AF8FFZAZYNE5  0000000078     5.0  1092182400
1          AH2L9G3DQHHAJ  0000000116     4.0  1019865600
2         A2IIIDRK3PRRZY  0000000116     1.0  1395619200
3         A1TADCM7YWPQ8M  0000000868     4.0  1031702400
4          AWGH7V0BDOJKB  0000013714     4.0  1383177600
...                  ...         ...     ...         ...
82677126  A349MZLDSQAIQC  BT00IU6O8K     3.0  1375401600
82677127  A1NFN9O4A2Q92V  BT00IU6O8K     5.0  1375142400
82677128  A1DA2L6U0AFMLL  dp-g310/do     5.0  1195948800
82677129   AIDHT6KXOL64T  SMLRBIMX03     5.0  1355788800
82677130  A161I1MHX2G2BR  SMLRBIMX03     4.0  1364860800

[82677131 rows x 4 columns]

In [6]:
amazon_df['user'].value_counts()

A14OJS0VWMOSWO    44557
AFVQZQ8PW0L       29044
A328S9RN3U5M68     9773
A9Q28YTLYREO7      9243
A2EDZH51XHFA9B     6431
                  ...  
A1XI67YEJNA6WK        1
A3GEDU3AJXFIRS        1
A3FJ134CQ53ARF        1
ASIOBI8B1Q7OE         1
AQCNH5KG3W1SW         1
Name: user, Length: 21176522, dtype: int64

There are 21,176,522 different users who wrote reviews in this dataset.

In [7]:
amazon_df['item'].value_counts()

B0054JZC6E    25368
B00FAPF5U0    24024
B009UX2YAC    23956
0439023483    21398
030758836X    19867
              ...  
1561640972        1
B0000015IQ        1
B0019GO1JU        1
B00CBO4S92        1
B001RLZ6Y6        1
Name: item, Length: 9874211, dtype: int64

There are 9,874,211 unique products represented in this dataset

In [8]:
amazon_df[['item','rating']].to_csv(r'rough_data\amazon_dataset\amzn_all_items_simplified.csv', index=False)

I created a new .csv file for the combined data, named 'amzn_all_items_simplified.csv', where I got rid of unneeded columns. This will make the data much more useable on my computer. Future work will be done with that file